test/test_nn.py

```diff
@@ -1890,6 +1890,23 @@ def test_embedding_functional(self):
         res_F = F.embedding(a, embeddings)
         self.assertEqual(res_old, res_F)
 
+    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    @repeat_test_for_types([torch.float, torch.half])
+    def test_softmax_dtype(self, dtype=torch.float):
+        input = torch.tensor([32, 100], device="cuda", dtype=dtype).uniform_()
```

Comment:
nit: this is equivalent to `torch.rand(32, 100, device='cuda', dtype=dtype, requires_grad=True)`